Rework static/shared variants into a configure-time build-switch (#688)

t-b opened the pull request:
Shared/static switching on windows

configure/CMakeLists.txt: Don't link against msvcstub anymore on windows

We don't link against that library from omniorb anymore as it is only needed as a workaround for using ominORB DLLs under MSVC++ 5.0 (due to destructor inlining issues).

Co-authored-by: Marius Elvert <marius.elvert@softwareschneiderei.de>

Close #826

ltjax committed:
Rework static/shared variants into a configure-time build-switch (#688)

Shared/static switching on windows

configure/CMakeLists.txt: Don't link against msvcstub anymore on windows

We don't link against that library from omniorb anymore as it is only needed as a workaround for using ominORB DLLs under MSVC++ 5.0 (due to destructor inlining issues).

Co-authored-by: Marius Elvert <marius.elvert@softwareschneiderei.de>

beenje commented:
I see that CI failed but I managed to build this branch on windows using conda-build (with only a patch to rename the libzmq and pthread libraries according to existing conda packages). I didn't run any tests.

t-b commented:
@beenje Thanks for trying. I think I fixed the build error. It was in appveyor.yml.

beenje commented:
> @beenje Thanks for trying. I think I fixed the build error. It was in appveyor.yml.

OK. If that's the only file you modified, that will not change anything for the conda build. I can still double check when the CI passes if you want.

t-b commented:
@mliszcz @bourtemb Ready.

mliszcz commented on configure/cpack_win.cmake:
```diff
@@ -1,33 +1,20 @@
+set(COMPILER_TAG ${CMAKE_VS_PLATFORM_TOOLSET})
```
For some reason `CMAKE_VS_PLATFORM_TOOLSET` was not used with `MSVC90`, but now it is. Are you sure it is defined? The documentation only mentions that: *VS 10 and above use MSBuild under the hood ...*

https://cmake.org/cmake/help/latest/variable/CMAKE_VS_PLATFORM_TOOLSET.html

mliszcz replied:
Yes, it is undefined. Just see the logs:
```
CPack: - package: C:/projects/static_build/libtango_9.3.4__x64_static_release.msi generated
```

Other jobs print version here:
```
CPack: - package: C:/projects/static_build/libtango_9.3.4_v141_x64_static_debug.msi generated.
```

t-b replied:
Thanks for carerful checking. I'll find a workaround.

t-b replied:
@mliszcz Done.

t-b committed:
configure/CMakeLists.txt: Handle CMAKE_VS_PLATFORM_TOOLSET for MSVC90

The last commit was from the tango-9-lts branch which did not had
support for MSVC90.

In this branch we still have support for MSVC90 and
CMAKE_VS_PLATFORM_TOOLSET is not set for this generator.

So we need to manually set it.

mliszcz reviewed (approved):
Looks good now! First msvc9 job just completed with correct package name:
```
CPack: - package: C:/projects/static_build/libtango_9.3.4_v90_x86_static_release.msi generated.
```

One comment though: CMAKE_VS_PLATFORM_TOOLSET variable is [supposed to be set by CMake](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#variables-that-provide-information) to give use some information, not the other way around. I wouldn't overwrite that one but rather COMPILER_TAG which is owned by us.
But since the current solution works, and it's for 9.3 and for msvc9 only, I think it won't do any harm.
